trustfall_core/test_data/tests/valid_queries/fold_count_filter_on_a_tag.graphql.ron

```diff
@@ -0,0 +1,38 @@
+TestGraphQLQuery (
+    // This test ensures that we do not ignore filters that should
+    // disable an optimization in which we only compute part of a
+    // fold.
+    //
+    // This test should result in no results. However, if we partially
+    // compute the fold ignoring the `>=` filter, we will get a result
+    // of `{ "value": 6 }`.
+    schema_name: "numbers",
+    query: r#"
+{
+    Number(min: 6, max: 6) {
+        ... on Composite {
+
+            value @output
+
+            predecessor { # five
+                predecessor { # four
+                    predecessor { # three
+                        predecessor { # two
+                            predecessor { # one
+                                value @tag(name: "one")
+                            }
+                        }
+                    }
+                }
+            }
+
+            primeFactor @fold @transform(op: "count")
+                        @filter(op: "<", value: ["%one"])
+                        @filter(op: ">=", value: ["$zero"])
+        }
+    }
+}"#,
```

Comment:
This query is quite long, which makes its test files large (the trace alone is 700+ lines) and also would make debugging unnecessarily complex.

Here's a simpler variant to consider. I've also swapped the order of the filters so that the passing one gets evaluated first (because right now we run them in the same order as in the query):
```suggestion
{
    Two {
        value @tag(name: "two")

        multiple(max: 3) {
            value @filter(op: "=", value: ["$six"]) @output

            primeFactor @fold @transform(op: "count")
                        @filter(op: ">=", value: ["$zero"])
                        @filter(op: "<", value: ["%two"])
        }
    }
}"#,
```

What do you think?

Reply:
You're right, this is a good change. I didn't even think to check the trace file size, but it's now down by 300 lines to only 400 after your change